### DISCUSSION TITLE
[Unstable] Opiates' burn healing has no practical applications, but does confuse the bots

### DISCUSSION THREAD

**Discussion by NotWendy** [FakeFishGames/Barotrauma#13785] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

In the current Unstable version of the game morphine heals 0.1 burns per second, for 20 seconds, which amounts to 2 burn healing.
That's nothing. It's never relevant, you will never use morphine to treat burns. But bots will, because morphine has a positive suitability for burn treatment from their perspective, however low it is.

So the question is why? Why should morphine heal a minuscule amount of burns, the affliction which goes away on its own over time and can be treated manually with bandages, which are very common? If the only effect it has on the gameplay is making bots have a harder time understanding morphine's applications, why not remove the burn healing completely?

Other opiates (opium, fentanyl) have the same problem, with slightly different numbers.

### Reproduction steps

1. Apply 40 burns or so to yourself (while being otherwise healthy) and ask a medic who has no access to bandages/plastiseal/glue to heal you.
2. Watch them inject you with a dose of morphine/opium/fentanyl before giving up.
3. Enjoy recovering from 2 additional points of burns over the next 20 seconds.

### Bug prevalence

Happens every time I play

### Single player or multiplayer?

Happens in both single player and multiplayer

### Version

v1.4.3.0 (unstable)

### Which operating system did you encounter this bug on?

Windows

**Comment by SomeRandomNoobKekeke** [comment]
The reason why I kept morphine in the toxin cabinet

**Comment by Regalis11** [comment]
> So the question is why? Why should morphine heal a minuscule amount of burns, the affliction which goes away on its own over time and can be treated manually with bandages, which are very common? If the only effect it has on the gameplay is making bots have a harder time understanding morphine's applications, why not remove the burn healing completely?

In my opinion these are two unrelated things: the medicines that bots use are configured manually, what and how much the medicine actually treats doesn't affect it. So I would rather ask: why not? Even though the minuscule healing of burns very rarely has any practical applications, it doesn't cause any problems either, and theoretically there could be some rare situations in which the burn reduction does have some practical implications (like let's say, if you have some affliction that makes opiates much more powerful and makes the burn reduction meaningful).

Rather than removing the burn healing, I'd just make the bots not consider opiates viable treatments for burns.

**Reply by NotWendy** [reply]
I've thought about that and yea, that's true. Removing the line which gives opiates any suitability score for burns would work.
The reason I suggested removing burn healing from opiates entirely is to avoid confusion among both modders and people working on the medical system in the future. We're creating an exception by making bots not take the burn healing effect into consideration. All for the sake of keeping an effect which never comes into play.
Someone stacking medical buffs and using opiates to heal burns just sounds like a completely made-up situation to me, because of how pitifully low the base healing is. But yes, keeping that effect isn't a big deal, at the end of the day.

**Reply by Regalis11** [reply]
> Someone stacking medical buffs and using opiates to heal burns just sounds like a completely made-up situation to me

Could be yeah, I'm just thinking these kinds of small things sometimes add up and lead to interesting emergent effects and strategies we wouldn't have thought about. Seems pretty unlikely in this case, but it could happen. :)

**Reply by NotWendy** [reply]
Looking forward to the anti-burn morphine meta. :BaroDev

**Comment by Regalis11** [comment]
Addressed in https://github.com/FakeFishGames/Barotrauma-development/pull/5203 by making the bots not consider opiates valid treatments for burns.
